src/main/scala/nl.knaw.dans.easy.bagstore/ArchiveStream.scala

```diff
@@ -45,12 +45,13 @@ case class EntrySpec(sourcePath: Option[Path], entryPath: String)
  * @param files the files in the TAR ball
  */
 class ArchiveStream(streamType: ArchiveStreamType, files: Seq[EntrySpec]) {
-  private val BLOCKSIZE = 1024 * 10 // TAR files are rounded up to blocks of 10K,
-
-  private val streamTypes = Map(
-    TAR -> ArchiveStreamFactory.TAR,
-    ZIP -> ArchiveStreamFactory.ZIP)
 
+  implicit private def toArchiveStreamFactory(streamType: ArchiveStreamType.Value): String = {
+    streamType match {
+      case TAR => ArchiveStreamFactory.TAR
+      case ZIP => ArchiveStreamFactory.ZIP
+    }
+  }
```

Comment:
@rvanheest: Why is this better?

Reply:
now the conversion is done automatically for you. Where it expects a `String`, you can just feed your enum value.

Reply:
maybe, now that I think of it, you could better move it to the `ArchiveStreamType` object, just as the other, similar conversions. In that way you keep all conversions close together. Then, when you want to add another type (say 7z) you only have to change a couple of conversions in one file, thereby reducing the chance of forgetting a conversion...